lua/nvim-tree/lib.lua

```diff
@@ -86,6 +87,32 @@ function M.get_last_group_node(node)
   return node
 end
 
+---Group empty folders
+---@param node Node
+---@return Node[]
+function M.group_empty_folders(node)
+  local is_root = not node.parent
+  local child_folder_only = explorer_node.has_one_child_folder(node) and node.nodes[1]
+  if M.group_empty and not is_root and child_folder_only then
+    node.group_next = child_folder_only
+    local ns = M.group_empty_folders(child_folder_only)
+    node.nodes = ns or {}
+    return ns
+  end
+  return node.nodes
+end
+
+---Group empty folders
```

Comment:
Group->ungroup and longer description please.

Reply:
sure,  will resolve the comments and working on option 2

Reply:
Just pushed a new version and implemented option 2, please take a look if you are free